qlib/utils/__init__.py

```diff
@@ -742,3 +744,55 @@ def load_dataset(path_or_obj):
     elif extension == ".csv":
         return pd.read_csv(path_or_obj, parse_dates=True, index_col=[0, 1])
     raise ValueError(f"unsupported file type `{extension}`")
+
+
+def set_config(config_c, default_conf="client", **kwargs):
+
+    config_c.reset()
+
+    _logging_config = config_c.logging_config
+    if "logging_config" in kwargs:
+        _logging_config = kwargs["logging_config"]
+
+    # set global config
+    if _logging_config:
+        set_log_with_config(_logging_config)
+
+    # FIXME: this logger ignored the level in config
+    logger = get_module_logger("Initialization", level=logging.INFO)
+    logger.info(f"default_conf: {default_conf}.")
+
+    config_c.set_mode(default_conf)
+    config_c.set_region(kwargs.get("region", config_c["region"] if "region" in config_c else REG_CN))
+
+    for k, v in kwargs.items():
+        if k not in config_c:
+            logger.warning("Unrecognized config %s" % k)
+        config_c[k] = v
+
+    config_c.resolve_path()
+
+    if not (config_c["expression_cache"] is None and config_c["dataset_cache"] is None):
+        # check redis
+        if not can_use_cache():
+            logger.warning(
+                f"redis connection failed(host={config_c['redis_host']} port={config_c['redis_port']}), cache will not be used!"
+            )
+            config_c["expression_cache"] = None
+            config_c["dataset_cache"] = None
+
+
+def config_based_on_c(config_c):
```

Comment:
Will it be better if this is a static methods in  `class Config`

Reply:
The name of the function could be greatly simplified if it is a static methods in `class Config`

Reply:
You can even change `config_c` to `self`